build-aux/snap/snapcraft.yaml

```diff
@@ -28,172 +29,294 @@ license: GPL-3.0
+      cp -a apparmor.d/tunables "${CRAFT_PART_INSTALL}/usr/lib/snapd/apparmor.d"
+    prime:
+      # We are statically linked
+      - -usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/**
+      - -usr/include
+    <<:
+      - *dynamic-linker
+    override-prime: |
+      craftctl default
+      python3 "${CRAFT_PROJECT_DIR}/build-aux/snap/local/patch-dl.py" "/snap/snapd/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/${DYNAMIC_LINKER}"
+
+  snapd:
+    plugin: nil
+    source: .
+    build-snaps:
+      - go/1.18/stable
+    after:
+      - apparmor
+    # these packages are needed to call mkversion.sh in override-pull, all other
+    # dependencies are installed using apt-get build-dep
+    build-packages:
+      - git
+      - dpkg-dev
+      - autoconf
+      - autoconf-archive
+      - automake
+      - xfslibs-dev
+      - libudev-dev
+      - libcap-dev
+      - libseccomp-dev
+      - pkg-config
+    override-pull: |
+      craftctl default
+      # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
+      craftctl set version="$(./mkversion.sh --output-only)"
+      ./get-deps.sh --skip-unused-check
+    override-build: |
+      ./mkversion.sh
+      cd "${CRAFT_PART_BUILD}/cmd"
+
+      autoreconf -fvi
+      ./configure --prefix=/usr --libexec=/usr/lib/snapd --without-unit-tests --enable-static-libapparmor --disable-host-binaries
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}"
+      make -j1 DESTDIR="${CRAFT_PART_INSTALL}" install
+
+      cd "${CRAFT_PART_BUILD}"
+
+      # Work-around for snapcraft not working with git worktrees
+      # We detect it with .git being a file rather than a directory
+      if [ -f .git ]; then
+        EXTRA_GO_FLAGS="-buildvcs=false"
+      fi
+      CMDS=(bin/snap
+            lib/snapd/snapd
+            lib/snapd/snap-exec
+            lib/snapd/snap-failure
+            lib/snapd/snap-fde-keymgr
+            lib/snapd/snap-preseed
+            lib/snapd/snap-recovery-chooser
+            lib/snapd/snap-repair
+            lib/snapd/snap-seccomp
+            lib/snapd/snap-update-ns
+            lib/snapd/snapctl
+            lib/snapd/snapd-apparmor)
+      for cmd in "${CMDS[@]}"; do
+        case "${cmd}" in
+           # FIXME: some binaries need to be run confined in apps. But
+           # instead we should allow apps to access dynamic linkers
+           # and libraries from snapd.
+           lib/snapd/snap-exec|lib/snapd/snapctl)
+             export CGO_ENABLED=0
+             GO_LD_FLAGS=()
+             ;;
+           lib/snapd/snap-update-ns)
+             export CGO_ENABLED=1
+             GO_LD_FLAGS=(--ldflags '-extldflags "-static"')
+             ;;
+           *)
+             export CGO_ENABLED=1
+             GO_LD_FLAGS=()
+             ;;
+        esac
+
+        case "${cmd}" in
+          bin/snap)
+            TAGS=(nomanagers)
+            case "$(craftctl get version)" in
+              1337.*)
+                TAGS+=(withtestkeys faultinject)
```

Comment:
This was enabled by https://github.com/snapcore/snapd/commit/f61579386bf81f05d1f71c1c7098a759db64990f

But I do not see how it would actually get used. Was it for `snap-bootstrap`? But even `snap-bootstrap` does not seem to use it.

Reply:
Hm ok, let's leave it for consistency with deb packaging. There is no cost associated with it anyway unless there are calls to the package, which there shouldn't be outside of snapd.

Reply:
OK, I will remove the commit that removed it then.

Reply:
I have force pushed to remove that commit.